### PR TITLE
[JBPM-9799] - Create auto-generated PIM DDL scripts by Hibernate

### PR DESCRIPTION
**Pull request by afalhambra**
**[JBPM-9799](https://issues.redhat.com/browse/JBPM-9799)**: Create auto-generated PIM DDL scripts by Hibernate

### CODE REVIEW

File: process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java
Comment (ruromero) [inline process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java:23]:
```suggestion
    private static final Logger LOGGER = LoggerFactory.getLogger(EntityManagerProducer.class);
```

File: process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java
Comment (ruromero) [inline process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java:1]: Missing license

File: process-migration-service/src/test/java/org/kie/processmigration/persistence/ScriptPersistenceBaseTest.java
Comment (ruromero) [inline process-migration-service/src/test/java/org/kie/processmigration/persistence/ScriptPersistenceBaseTest.java:47]:
```suggestion
    private static final Logger LOGGER = LoggerFactory.getLogger(ScriptPersistenceBaseTest.class);
```

File: process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java
Comment (ruromero) [inline process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java:21]: Has this class been added only for the tests requirements?

**Review by ruromero — CHANGES_REQUESTED** [review]
Looks great @afalhambra !

**Comment by ruromero** [comment]
Why isn't this PR filed against https://github.com/kiegroup/process-migration-service ?

File: process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java
Comment (afalhambra) [reply process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java]: done

File: process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java
Comment (afalhambra) [reply process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java:17]: done

File: process-migration-service/src/test/java/org/kie/processmigration/persistence/ScriptPersistenceBaseTest.java
Comment (afalhambra) [reply process-migration-service/src/test/java/org/kie/processmigration/persistence/ScriptPersistenceBaseTest.java]: done

File: process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java
Comment (afalhambra) [reply process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java:37]: No - Since we want to allow customers to either auto-generate PIM schema through Hibernate or create it based on the DDL scripts provided, we need to set the Hibernate properties (`hibernate.hbm2ddl.auto` in this case) dynamically. Thus, if you set these properties as a system properties, they will be taken into account when creating the entity manager factory and overwrite the ones defined in the persistence.xml file

See: 
https://github.com/kiegroup/droolsjbpm-integration/blob/ba5039e6683c2faac2ae632c03bf29506e260df0/process-migration-service/src/main/java/org/kie/processmigration/service/EntityManagerProducer.java#L56

**Comment by afalhambra** [comment]
> Why isn't this PR filed against https://github.com/kiegroup/process-migration-service ?

I was not aware of such repo - should I raise a PR against that repo and close this one?

**Review by ruromero — APPROVED** [review]
Got it, thanks! LGTM

**Comment by ruromero** [comment]
I think you should, WDYT @mareknovotny ?

**Comment by mareknovotny** [comment]
@ruromero was already the process-migration-service module removed from this repository? I didn't see any PR but i was on PTO.

So i would recommend to wait until we setup CI to include the new repository https://github.com/kiegroup/process-migration-service and then remove it from this repository and announce new repository for the rest of the team on our ML.
Tagging @mbiarnes and @Ginxo for https://issues.redhat.com/browse/BXMSPROD-1387

**Comment by afalhambra** [comment]
> @ruromero was already the process-migration-service module removed from this repository? I didn't see any PR but i was on PTO.
> 
> So i would recommend to wait until we setup CI to include the new repository https://github.com/kiegroup/process-migration-service and then remove it from this repository and announce new repository for the rest of the team on our ML.
> Tagging @mbiarnes and @Ginxo for https://issues.redhat.com/browse/BXMSPROD-1387

@mareknovotny - just to confirm then, so for 7.12.x release, we should be using the new repo location https://github.com/kiegroup/process-migration-service once you have setup CI to include it. Thanks!

**Comment by Ginxo** [comment]
> @ruromero was already the process-migration-service module removed from this repository? I didn't see any PR but i was on PTO.
> 
> So i would recommend to wait until we setup CI to include the new repository https://github.com/kiegroup/process-migration-service and then remove it from this repository and announce new repository for the rest of the team on our ML.
> Tagging @mbiarnes and @Ginxo for https://issues.redhat.com/browse/BXMSPROD-1387

@mareknovotny 

https://github.com/kiegroup/process-migration-service/pull/1
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1694
https://github.com/kiegroup/kie-jenkins-scripts/pull/1036

are ready to be reviewed/merged

**Comment by mareknovotny** [comment]
could you resend this PR into new repository @afalhambra ? I merged now the all CI and repository-list.txt so we can try if it works as expected and also do not want to sync manually new repository as additional step. I will send PR for removal of PIM module immediately

**Comment by mareknovotny** [comment]
removed PIM is in https://github.com/kiegroup/droolsjbpm-integration/pull/2541

**Comment by mareknovotny** [comment]
the #2541 were merged now you need to send the changes to https://github.com/kiegroup/process-migration-service

**Pull request closed without merging**
